### PR TITLE
DOCS-2432 Wallarm mode directive in specific deployments

### PR DESCRIPTION
**Pull request by MaximBashurov**

### CODE REVIEW

File: docs/latest/admin-en/configure-wallarm-mode.md [hunk 1]
```diff
@@ -90,6 +90,78 @@ In this example, the filtration modes are defined for the resources as follows:
       2. The `block` mode is applied to the requests sent to the virtual server `SERVER_C` that contain the `/main/login` path.
       3. The `safe_blocking` mode is applied to the requests sent to the virtual server `SERVER_C` that contain the `/main/reset-password` path.
 
+**Setting `wallarm_mode` in specific deployments**
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configure-wallarm-mode.md:93]: this guide is super overloaded. is not a way. the above description practically duplicates the one from the wallarm_mode directive description in the directive list. examples of usage can be skipped - we provide them in the deployment guides

File: docs/latest/admin-en/configure-wallarm-mode.md [hunk 2]
```diff
@@ -32,73 +32,100 @@ The filtration mode can be configured in the following ways:
 
 Priorities of the filtration mode configuration methods are determined in the [`wallarm_mode_allow_override` directive](#setting-up-priorities-of-filtration-mode-configuration-methods-using-wallarm_mode_allow_override). By default, the settings specified in Wallarm Console have a higher priority than the `wallarm_mode` directive regardless of its value severity.
 
-### Specifying the filtration mode in the `wallarm_mode` directive
+### Configuration file `wallarm_mode` directive
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configure-wallarm-mode.md:35]: ?

Comment (AnastasiaTWW) [reply docs/latest/admin-en/configure-wallarm-mode.md:35]: not clear. "The `wallarm_mode` directive" maybe?

File: docs/latest/admin-en/configure-wallarm-mode.md [hunk 3]
```diff
@@ -32,73 +32,100 @@ The filtration mode can be configured in the following ways:
...
+* `location`: the directives are only applied to the requests containing that particular path.
 
 If different `wallarm_mode` directive values are defined for the `http`, `server`, and `location` blocks, the most local configuration has the highest priority.
 
-**The `wallarm_mode` directive usage example:**
+**Setting `wallarm_mode` in specific deployments**
 
-```bash
-http {
+Peculiarities on how the `wallarm_mode` directive is set in the specific deployments are described below. Note that the values and their meanings are the same as for the `wallarm_mode` directive.
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configure-wallarm-mode.md:50]: if you list all the deployment otions below, then list them all. no aio here, cloud images. the above info on nginx contexts is relevant only for some deployment options, as far as I could see

File: docs/latest/admin-en/configure-wallarm-mode.md [hunk 4]
```diff
@@ -32,73 +32,100 @@ The filtration mode can be configured in the following ways:
...
+* `server`: the directives are applied to the requests sent to the virtual server.
+* `location`: the directives are only applied to the requests containing that particular path.
 
 If different `wallarm_mode` directive values are defined for the `http`, `server`, and `location` blocks, the most local configuration has the highest priority.
 
-**The `wallarm_mode` directive usage example:**
+**Setting `wallarm_mode` in specific deployments**
 
-```bash
-http {
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configure-wallarm-mode.md:51]: Docker and ENvoy - they both are about Docker. the level of logical grouping is not clear

File: docs/latest/admin-en/configure-wallarm-mode.md [hunk 5]
```diff
@@ -32,73 +32,100 @@ The filtration mode can be configured in the following ways:
...
-* `server`: the directives inside the `server` block are applied to the requests sent to the virtual server.
-* `location`: the directives inside the `location` block are only applied to the requests containing that particular path.
+* `http`: the directives are applied to the requests sent to the HTTP server.
+* `server`: the directives are applied to the requests sent to the virtual server.
+* `location`: the directives are only applied to the requests containing that particular path.
 
 If different `wallarm_mode` directive values are defined for the `http`, `server`, and `location` blocks, the most local configuration has the highest priority.
 
-**The `wallarm_mode` directive usage example:**
+**Setting `wallarm_mode` in specific deployments**
```
Comment (AnastasiaTWW) [inline docs/latest/admin-en/configure-wallarm-mode.md:49]: also, mention that settings wallarm_mode to one of the values from the table makes sense only for inline deployments. for oob, only the monitoring mode is relevant

Comment (AnastasiaTWW) [reply docs/latest/admin-en/configure-wallarm-mode.md:49]: and for oob there are also deployment options like ebpf that allow to control the mode in a dedicated way

File: docs/latest/admin-en/configure-wallarm-mode.md (hunk 3 above)
Comment (MaximBashurov) [reply docs/latest/admin-en/configure-wallarm-mode.md:50]: - I do not list all deployment options, only "specific"
- "the above info on nginx contexts..." - did not understand that

File: docs/latest/admin-en/configure-wallarm-mode.md (hunk 4 above)
Comment (MaximBashurov) [reply docs/latest/admin-en/configure-wallarm-mode.md:51]: I cannot imagine a better way of groping here. Could you advise something?

File: docs/latest/admin-en/configure-wallarm-mode.md (hunk 5 above)
Comment (MaximBashurov) [reply docs/latest/admin-en/configure-wallarm-mode.md:49]: Regarding eBPF, I found nothing connected to the filtering mode in the Helm Chart Values article. Moreover, as this is an OOB solution, how can it have the filtering mode different from monitoring?

**Pull request merged**